ipatests/test_integration/test_installation.py

```diff
@@ -847,3 +847,34 @@ def test_replicamanage_del(self):
 
         assert_text = 'Deleted IPA server "%s"' % self.replicas[0].hostname
         assert assert_text in cmd_output.stdout_text
+
+
+class TestInstallReplicaAgainstSpecificServer(IntegrationTest):
+    """Installation of replica against a specific server
+
+    Test to check replica install against specific server. It uses master and
+    replica1 without CA and having custodia service stopped. Then try to
+    install replica2 from replica1 so that replica2 will fetch secrets from
+    master as custodia service is not running on replica1.
+
+    related ticket: https://pagure.io/freeipa/issue/7566
+    """
+
+    num_replicas = 2
+
+    def test_replica_install_against_server(self):
+        tasks.install_master(self.master)
+
+        # install replica1 without CA
+        tasks.install_replica(self.master, self.replicas[0], setup_ca=False)
```

Comment:
I would install the replica with DNS, in order to make sure that it's the lack of CA that forces the use of master, and not the lack of DNS.